docker/puppetdb/Dockerfile

```diff
@@ -56,4 +56,15 @@ EXPOSE 8080 8081
 ENTRYPOINT ["dumb-init", "/docker-entrypoint.sh"]
 CMD ["foreground"]
 
+# The start-period is just a wild guess how long it takes PuppetDB to come
+# up in the worst case. The other timing parameters are set so that it
+# takes at most a minute to realize that PuppetDB has failed.
+HEALTHCHECK --interval=10s --timeout=10s --retries=6 \
+  --start-period=5m CMD \
```

Comment:
Depending on what's wanted, this might not be sufficient.  i.e. pdb's initial status is `starting`, and it only shifts to `running` when it leaves maintenance mode, and that can take a while if there's a big migration, and a *very* long time if sync is enabled, and the server is sufficiently far behind (hours, even a day or more for very large installs -- though the latter should hopefully get better in the not too distant future).

cf. https://puppet.com/docs/puppetdb/5.1/api/status/v1/status.html

Reply:
What the healthcheck is trying to achieve is to signal that PuppetDB is ready for business; in particular, 'healthy' should translate to 'ready to accept catalogs and reports'. From the docs, it sounds like we do have to wait for the state to go to 'running'.

What do users usually do to make sure their Puppet setup is ready in the face of long running ingestion times ?

Reply:
Yes, you wait for a "running" state, then this check fires every 10 seconds (based on this config) for the rest of the runtime of the container to make sure it stays healthy.

Reply:
I believe as far as puppetdb is concerned, at least, they just wait.  Charlie suggested that the existing init script may have a very long timeout (hours?) to allow for expensive migrations (and/or presumably, initial blocking HA syncs).  That could occur because puppetdb had to introduce an expensive migration in a new version, or because the particular site has a *lot* of data, or slow storage, or...